.github/workflows/test.yml

```diff
@@ -157,6 +158,8 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
+          ## Temp testing -- not a permanent change
+          python -m pip install -r requirements-dev.txt -c constraints-dev.txt --use-deprecated=legacy-resolver
```

Comment:
This PR fixes two bugs in a single commit.

It will be difficult to revert the original click 8.1 constraint change. Can you please separate them so it will be easier to revert this?

Reply:
I'm not sure I understand. Can't we just create a new PR to undo this change once we update the source code to be compatible with the newest version of click?

Reply:
The switch to `legacy-resolver` is unrelated to the `click` change, right? If it is unrelated, it will be clearer to have these two bug fixes as two separate commits.

Reply:
We should not switch to legacy resolver, it allows to install incompatible version.

Reply:
(btw I support Ken's proposal to separate changes into separate commits that we can revert with `git revert` - manually reverting these is a chore)